tests/test_docker_cmdline.py

```diff
@@ -0,0 +1,260 @@
+        pass
+    return (param_list, args_left)
+
+
+def flatten(container: Iterable) -> List:
+    return list(itertools.chain.from_iterable(container))
+
+
+def assert_docker_std_cmdline(run_args: List[str], sudo_outside: bool = False) -> List[str]:
+    expected_args = []
+    if sudo_outside:
+        expected_args.append('sudo')
+    expected_args.extend(['docker', 'run', '--rm'])
+    assert expected_args == run_args[:len(expected_args)]
+    return run_args[len(expected_args):]
+
+
+def assert_docker_image_and_cmd_inside_docker(run_args: List[str], expected_docker_image: str, expected_cmd_inside_docker: List[str]) -> List[str]:
+    run_args_cmd_inside_docker = run_args[-len(expected_cmd_inside_docker):]
+    assert expected_cmd_inside_docker == run_args_cmd_inside_docker
+    assert expected_docker_image == run_args[-(len(expected_cmd_inside_docker) + 1)]
+    return run_args[:-(len(expected_cmd_inside_docker) + 1)]
+
+
+def assert_workdir_param(run_args: List[str], expected_workdir_path: Path) -> List[str]:
+    workdir_params, args_left = split_single_cmdline_param('-w', run_args, include_value=True)
+    assert ['-w', str(expected_workdir_path)] == workdir_params
+    return args_left
+
+
+def assert_hostname_param(run_args: List[str], expected_hostname: str) -> List[str]:
+    hostname_params, args_left = split_single_cmdline_param('--hostname=', run_args)
+    assert ['--hostname={}'.format(expected_hostname)] == hostname_params
+    return args_left
+
+
+def assert_volume_params(run_args: List[str], expected_volume_mappings: List[Tuple[str, str]]) -> List[str]:
+    volume_params, args_left = split_single_cmdline_param('-v', run_args, include_value=True)
+    vol_mapping_values = ['{}:{}'.format(vm[1], vm[0]) for vm in expected_volume_mappings]
+    expected_volume_params = flatten(zip(['-v'] * len(expected_volume_mappings), vol_mapping_values))
+    assert expected_volume_params == volume_params
+    return args_left
+
+
+def assert_interactive(run_args: List[str], expected_interactive: bool) -> List[str]:
+    interactive_param, args_left = split_single_cmdline_param('-i', run_args)
+    expected_interactive_param = ['-i'] if expected_interactive else []
+    assert expected_interactive_param == interactive_param
+    return args_left
+
+
+def assert_env_params(run_args: List[str], expected_env_values: List[str]) -> List[str]:
+    env_params, args_left = split_single_cmdline_param('-e', run_args, include_value=True)
+    expected_env_params = flatten(zip(['-e'] * len(expected_env_values), expected_env_values))
+    assert expected_env_params == env_params
+    return args_left
+
+
+def std_assert_hostname_param(args_left):
+    return assert_hostname_param(args_left, 'dog_docker')
+
+
+def std_assert_volume_params(args_left):
+    return assert_volume_params(args_left, [('/tmp', '/tmp'), ('/home/test_home/.ssh:ro', '/home/test_home/.ssh'), ('/home/test_home/.p4tickets:ro', '/home/test_home/.p4tickets')])
```

Comment:
I can not understand why /tmp is here - is that the default directory or something like that - it should be cwd - so it should be "mock'able" as anything.

Reply:
Could you try fixing the tests so they run here on GitHub Actions - i think that would make it much easier to see that everything checks out.